test/integration/oauth_ldap_test.go

```diff
@@ -0,0 +1,180 @@
+// +build integration,!no-etcd
+
+package integration
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/client"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+	testutil "github.com/openshift/origin/test/util"
+
+	"github.com/vjeantet/ldapserver"
+)
+
+func TestOAuthLDAP(t *testing.T) {
+	const (
+		providerName = "myldapprovider"
+
+		bindDN       = "o=company,uid=admin"
```

Comment:
Add some sort of randomization to this and the password.  I had a bad experience and now I'm afraid of hardcoded usernames and passwords like this in tests.